src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,146 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { MdArrowForwardIos } from 'react-icons/md';
+
+function renderPagination(
+  pages: number,
+  currentPage: number,
+  onPageChange: (newPage: number) => void,
+  pageClass: string,
+  currentpageClass: string,
+): React.ReactNode[] {
+  const pagination: React.ReactNode[] = [];
+
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === pages;
+
+  const addPageLink = (pageNumber: number, className: string): void => {
+    pagination.push(
+      <a
+        onClick={() => onPageChange(pageNumber)}
+        className={className}
+        key={pageNumber}
+      >
+        {pageNumber}
+      </a>,
+    );
+  };
+
+  const addEllipsis = (key: string): void => {
+    pagination.push(
+      <span className={pageClass} key={key}>
+        ...
+      </span>,
+    );
+  };
+
+  addPageLink(1, currentPage === 1 ? currentpageClass : pageClass);
+
+  if (pages > 7) {
+    if (currentPage > 2 && !isFirstPage) {
+      addEllipsis('startEllipsis');
+    }
+
+    if (isFirstPage) {
+      for (let i = currentPage + 1; i < Math.min(currentPage + 4, 4); i++) {
+        addPageLink(i, i === currentPage ? currentpageClass : pageClass);
+      }
+    } else if (isLastPage) {
+      for (let i = currentPage - 4; i < pages; i++) {
```

Comment:
What is the 4 here!? Give a name in a const for that MIN_LEFT_PAGES, etc...

Reply:
done